clients/ts/signalr/spec/HubConnection.spec.ts

```diff
@@ -271,6 +271,72 @@ describe("HubConnection", () => {
             expect(warnings).toEqual(["No client method with the name 'message' found."]);
         });
 
+        it("all handlers can be unnregistered with just the method name", async () => {
+            const warnings: string[] = [];
+            const connection = new TestConnection();
+            const hubConnection = new HubConnection(connection);
+
+            connection.receiveHandshakeResponse();
+
+            let count = 0;
+            const handler = () => { count++; };
+            hubConnection.on("inc", handler);
+            hubConnection.on("inc", handler);
+
+            connection.receive({
+                arguments: [],
+                invocationId: "0",
+                nonblocking: true,
+                target: "inc",
+                type: MessageType.Invocation,
+            });
+
+            hubConnection.off("inc");
+
+            connection.receive({
+                arguments: [],
+                invocationId: "0",
+                nonblocking: true,
+                target: "inc",
+                type: MessageType.Invocation,
+            });
+
+            expect(count).toBe(2);
+        });
+
+        it("a single handler can be unnregistered with the method name and handler", async () => {
+            const warnings: string[] = [];
+            const connection = new TestConnection();
+            const hubConnection = new HubConnection(connection);
+
+            connection.receiveHandshakeResponse();
+
+            let count = 0;
+            const handler = () => { count++; };
+            hubConnection.on("inc", handler);
+            hubConnection.on("inc", handler);
+
+            connection.receive({
+                arguments: [],
+                invocationId: "0",
+                nonblocking: true,
+                target: "inc",
+                type: MessageType.Invocation,
+            });
+
+            hubConnection.off("inc", handler);
```

Comment:
This brings up an interesting question, if you call `off` with `handler` and have multiple `handler` registered, should it remove all instance of that callback?

Reply:
Hmm, I don't think so. Is that even a realistic scenario?

Reply:
IMO we should follow `addEventListener`'s lead, which is ignore registering the same function multiple times.

https://stackoverflow.com/questions/26127336/javascript-event-addeventlistener-registering-multiple-times-for-same-function

Reply:
Okay. I think that makes sense but maybe should be a different issue. Thoughts?

Reply:
It seems like it would be simple enough to do. Double check with Andrew or David that its behavior should change first.

Reply:
I agree. It means scanning the list to see if any of the existing items `===` the new item when adding, but I think it's OK. Adding handlers is a startup task. Go ahead and put that in this PR.

Reply:
You guys are the masters of feature creep

Reply:
It always happens to me too 🙁